### DISCUSSION TITLE
Tenacity timeouts

### DISCUSSION THREAD

**Discussion by kondera** [ag2ai/faststream#1362] [General]
Hey, how to go about adding exponential backoff timeouts with brokers? I see there's timeout that I can pass seconds (not really documented which I found odd). It would be nice to be able to pass timeouts and retries (like in tenacity) for brokers.

**Comment by Lancetnik** [comment]
I am not sure about `seconds` you talking about. Anyway, we can't impement such feature in framework if broker doesn't support it: there is no any sence to consume the message in memory, but do not pass it to user until timeout released. Also we can't gurantee that message we will consume from the queue: it can be the same one or another, so we can't wait any time before next message consuming.

The only way I can see - block your handler at error for `timeout` time to block process before next message consuming. But it sounds  ridiculous.

**Reply by kondera** [reply]
Oh, shoot, I didn't describe that well. I'm using RabbitMQ and deployed a small service in Kubernetes where RabbitMQ was reachable, but traffic was blocked by policies, causing it to hang up while trying to make a connection. I tried looking in the documentation for how to specify timeouts for creating connections and found it, but couldn't figure out what time unit I need to pass (seconds ;) ). And it got me thinking whether there's a way to use exponential backoff and retries (like with tenacity).

**Reply by Lancetnik** [reply]
Are you meaning for exponential timeout at connection only? It makes some sense, thanks!
